runtime/arc.js

```diff
@@ -159,7 +169,30 @@ class Arc {
     return s;
   }
 
-  connectParticleToView(particleHandle, particle, name, targetView) {
+  instantiate(recipe) {
+    assert(recipe.isResolved(), 'Cannot instantiate an unresolved recipe');
+    let {views, particles, slots} = recipe.mergeInfo(this._activeRecipe);
+    let recipeViewMap = new Map();
+    views.forEach(recipeView => {
```

Comment:
These views are only the views that need to be created after the merge, 
not all the views referenced by particles in recipe.